thrift/src/test/java/com/linecorp/armeria/it/metrics/PrometheusMetricsIntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.serviceAt("/thrift", THttpService.of((Iface) name -> {
+                if ("world".equals(name)) {
+                    return "success";
+                }
+                throw new IllegalArgumentException("bad argument");
+            }).decorate(
+                    PrometheusMetricRequestDecorator
+                            .decorateService(collectorRegistry,
+                                             MyMetricLabel.values(),
+                                             log -> defaultMetricName(log, "HelloService"))))
+              .serviceAt("/internal/prometheus/metrics-v2",
+                         new PrometheusExporterHttpService(collectorRegistry));
+        }
+    };
+
+    private static void makeRequest(String name) throws TException {
+        final Iface client = new ClientBuilder(server.uri(BINARY, "/thrift"))
+                .decorator(RpcRequest.class, RpcResponse.class,
+                           PrometheusMetricRequestDecorator
+                                   .decorateClient(collectorRegistry,
+                                                   MyMetricLabel.values(),
+                                                   log -> defaultMetricName(log, "HelloService")))
+                .build(Iface.class);
+        client.hello(name);
+    }
+
+    private static AggregatedHttpMessage makeMetricsRequest() throws ExecutionException,
+                                                                     InterruptedException {
+        final HttpClient client = Clients.newClient(ClientFactory.DEFAULT,
+                                                    "none+http://127.0.0.1:" + server.httpPort(),
+                                                    HttpClient.class);
+        return client.execute(HttpHeaders.of(HttpMethod.GET, "/internal/prometheus/metrics-v2")
+                                         .set(HttpHeaderNames.ACCEPT, "utf-8"))
+                     .aggregate().get();
+    }
+
+    private static Map<MyMetricLabel, String> defaultMetricName(RequestLog log,
+                                                                String serviceName) {
+        final RequestContext ctx = log.context();
+        final Object requestEnvelope = log.requestEnvelope();
+        final Object requestContent = log.requestContent();
+
+        final String path;
+        final String methodName;
+
+        if (requestContent instanceof RpcRequest) {
+            methodName = ((RpcRequest) requestContent).method();
+        } else if (requestEnvelope instanceof HttpHeaders) {
+            methodName = ((HttpHeaders) requestEnvelope).method().name();
+        } else {
+            methodName = log.method();
+        }
+
+        if (requestEnvelope instanceof HttpHeaders) {
+            path = ctx.path();
+        } else {
+            path = null;
+        }
+
+        return ImmutableMap.<MyMetricLabel, String>builder()
```

Comment:
Oh I see ``ImmutableSortedMap.copyOf`` will check for same order, and type, and just reuse the same instance
https://github.com/google/guava/blob/557649df91e5f8bab4d5f39d56c81bd47e134b06/guava/src/com/google/common/collect/ImmutableSortedMap.java#L337